docs/development/local_setup.md

```diff
@@ -127,6 +127,36 @@ The Gardener repository and all the above-mentioned tools (git, golang, kubectl,
 
 In case you have to create a new release or a new hotfix of the Gardener you have to push the resulting Docker image into a Docker registry. Currently, we are using the Google Container Registry (this could change in the future). Please follow the official [installation instructions from Google](https://cloud.google.com/sdk/downloads).
 
+## [Optional] Install GNU screen
+
+When screen is installed, you can easily start all local daemons using `make start-all`
+To install screen on *MacOS*
+
+```bash
+brew install screen
+```
+
+To install screen on *Debian/Ubuntu*
+
+```bash
+apt install screen
+```
+
+## NixOS/Nix support
+
+Gardner contains a `shell.nix` file for development purposes. All the required dependencies
+will be automatically installed when your run:
+
+```bash
+nix-shell
+```
+
+You can use this environment from a IDE with commands like:
+
+```bash
+nix-shell --command "make generate"
+```
```

Comment:
Could you point the advantages of this distro/shell?

Reply:
It is fully reproducible and package compilation and configuration is done by the same system. A user package configuration can also trigger that a package is compiled with different flags if necessary without any user interaction. You don't need NixOS to enjoy the power of nix, it can be installed on other operating systems.

https://nixos.org/guides/how-nix-works.html

Reply:
I linked in the docs to the nix page

Reply:
Well, we have https://github.com/gardener/gardener/issues/2143 exactly for this purpose.

Reply:
@vpnachev docker containers are hardly reproduce-able, you can only store your build images indefinitely and hope you never loose them. If there is any apt-update whatsoever, any curl download without check-sums, your build is just similar, not identical.